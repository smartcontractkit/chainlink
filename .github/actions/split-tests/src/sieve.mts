import {Tests} from "./types.mjs";

export function sieveSlowTests(tests: Tests, slowTestMatchers?: string[]) {
  const slowTests: Tests = [];
  const filteredTests: Tests = [];

  if (!slowTestMatchers) {
    return {slowTests, filteredTests: tests};
  }

  // If the user supplies slow test matchers
  // then we go through each test to see if we get a case sensitive match

  tests.forEach((t) => {
    const isSlow = slowTestMatchers.reduce(
      (isSlow, matcher) => t.includes(matcher) || isSlow,
      false
    );
    if (isSlow) {
      slowTests.push(t);
    } else {
      filteredTests.push(t);
    }
  });

  return {slowTests, filteredTests};
}
