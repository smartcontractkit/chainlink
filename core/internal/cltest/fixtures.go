package cltest

import (
	"encoding/json"
	"testing"

	"github.com/smartcontractkit/chainlink/core/store/models"

	"github.com/ethereum/go-ethereum/core/types"
	"github.com/stretchr/testify/require"
	"github.com/tidwall/gjson"
)

// JSONFromFixture create models.JSON from file path
func JSONFromFixture(t *testing.T, path string) models.JSON {
	return JSONFromBytes(t, MustReadFile(t, path))
}

// LogFromFixture create ethtypes.log from file path
func LogFromFixture(t *testing.T, path string) types.Log {
	value := gjson.Get(string(MustReadFile(t, path)), "params.result")
	var el types.Log
	require.NoError(t, json.Unmarshal([]byte(value.String()), &el))

	return el
}

// TxReceiptFromFixture create ethtypes.log from file path
func TxReceiptFromFixture(t *testing.T, path string) *types.Receipt {
	jsonStr := JSONFromFixture(t, path).Get("result").String()

	var receipt types.Receipt
	err := json.Unmarshal([]byte(jsonStr), &receipt)
	require.NoError(t, err)

	return &receipt
}
