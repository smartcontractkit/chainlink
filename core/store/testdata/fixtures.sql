INSERT INTO keys (address, json, created_at, updated_at, next_nonce) VALUES (
    E'\\x3cb8e3FD9d27e39a5e9e6852b0e96160061fd4ea',
    '{"address":"3cb8e3fd9d27e39a5e9e6852b0e96160061fd4ea","crypto":{"cipher":"aes-128-ctr","ciphertext":"7515678239ccbeeaaaf0b103f0fba46a979bf6b2a52260015f35b9eb5fed5c17","cipherparams":{"iv":"87e5a5db334305e1e4fb8b3538ceea12"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"d89ac837b5dcdce5690af764762fe349d8162bb0086cea2bc3a4289c47853f96"},"mac":"57a7f4ada10d3d89644f541c91f89b5bde73e15e827ee40565e2d1f88bb0ac96"},"id":"c8cb9bc7-0a51-43bd-8348-8a67fd1ec52c","version":3}',
    '2019-01-01',
    '2019-01-01',
    0
);
INSERT INTO users (email, hashed_password, token_secret, created_at, updated_at) VALUES (
    'apiuser@chainlink.test',
    '$2a$10$bbwErtZcZ6qQvRsfBiY2POvuY6D4lwj/Vxq/PcVAL6o64nRaPgaEa', -- hash of literal string 'password'
    '1eCP/w0llVkchejFaoBpfIGaLRxZK54lTXBCT22YLW+pdzE4Fafy/XO5LoJ2uwHi',
    '2019-01-01',
    '2019-01-01'
);
