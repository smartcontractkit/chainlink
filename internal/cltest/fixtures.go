package cltest

const (
	key3cb8e3fd9d27e39a5e9e6852b0e96160061fd4ea = `{"address":"3cb8e3fd9d27e39a5e9e6852b0e96160061fd4ea","crypto":{"cipher":"aes-128-ctr","ciphertext":"7515678239ccbeeaaaf0b103f0fba46a979bf6b2a52260015f35b9eb5fed5c17","cipherparams":{"iv":"87e5a5db334305e1e4fb8b3538ceea12"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"d89ac837b5dcdce5690af764762fe349d8162bb0086cea2bc3a4289c47853f96"},"mac":"57a7f4ada10d3d89644f541c91f89b5bde73e15e827ee40565e2d1f88bb0ac96"},"id":"c8cb9bc7-0a51-43bd-8348-8a67fd1ec52c","version":3}`
)
