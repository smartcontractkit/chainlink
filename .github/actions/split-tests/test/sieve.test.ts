import {sieveSlowTests} from "../src/sieve.mjs";
import {testArr} from "./fixtures.mjs";

describe("sieveSlowTests", () => {
  it("works", () => {
    expect(sieveSlowTests([])).toMatchSnapshot();
    expect(sieveSlowTests([], [])).toMatchSnapshot();
    expect(sieveSlowTests(["keepme"], [])).toMatchSnapshot();
    expect(sieveSlowTests(["keepme"])).toMatchSnapshot();
    expect(sieveSlowTests(testArr, [])).toMatchSnapshot();
    expect(sieveSlowTests(testArr, ["noself"])).toMatchSnapshot();
    expect(sieveSlowTests(testArr, ["ouch.test.ts"])).toMatchSnapshot();
    expect(sieveSlowTests(testArr, ["bo", "ouch.test.ts"])).toMatchSnapshot();
  });
});
