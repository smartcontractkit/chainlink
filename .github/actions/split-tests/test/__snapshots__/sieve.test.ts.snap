// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sieveSlowTests works 1`] = `
{
  "filteredTests": [],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 2`] = `
{
  "filteredTests": [],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 3`] = `
{
  "filteredTests": [
    "keepme",
  ],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 4`] = `
{
  "filteredTests": [
    "keepme",
  ],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 5`] = `
{
  "filteredTests": [
    "foo.test",
    "bar.test",
    "baz.test",
    "yup.test",
    "nope.test",
    "bonk.test",
    "bop.test",
    "ouch.test.ts",
  ],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 6`] = `
{
  "filteredTests": [
    "foo.test",
    "bar.test",
    "baz.test",
    "yup.test",
    "nope.test",
    "bonk.test",
    "bop.test",
    "ouch.test.ts",
  ],
  "slowTests": [],
}
`;

exports[`sieveSlowTests works 7`] = `
{
  "filteredTests": [
    "foo.test",
    "bar.test",
    "baz.test",
    "yup.test",
    "nope.test",
    "bonk.test",
    "bop.test",
  ],
  "slowTests": [
    "ouch.test.ts",
  ],
}
`;

exports[`sieveSlowTests works 8`] = `
{
  "filteredTests": [
    "foo.test",
    "bar.test",
    "baz.test",
    "yup.test",
    "nope.test",
  ],
  "slowTests": [
    "bonk.test",
    "bop.test",
    "ouch.test.ts",
  ],
}
`;
